pkg/registry/core/pod/strategy.go

```diff
@@ -60,7 +61,8 @@ func (podStrategy) NamespaceScoped() bool {
 func (podStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
 	pod := obj.(*api.Pod)
 	pod.Status = api.PodStatus{
-		Phase: api.PodPending,
+		Phase:    api.PodPending,
+		QOSClass: qos.InternalGetPodQOS(pod),
```

Comment:
i think we should have a test case for this in strategy_test.go

it also would be good if we had a e2e that verified qos class is set as well for each qos tier.  maybe in e2e pods.go?